examples/llm/utils/nats_queue.py

```diff
@@ -34,15 +31,7 @@ def __init__(
         nats_server: str = "nats://localhost:4222",
         dequeue_timeout: float = 1,
     ):
-        self.nats_url = nats_server
-        self._nc: Optional[NATS] = None
-        self._js: Optional[JetStreamContext] = None
-        # TODO: check if this is needed
-        # Sanitize stream_name to remove path separators
-        self._stream_name = stream_name.replace("/", "_").replace("\\", "_")
-        self._subject = f"{self._stream_name}.*"
-        self.dequeue_timeout = dequeue_timeout
-        self._subscriber: Optional[JetStreamContext.PullSubscription] = None
+        self.nats_q = RustNatsQueue(stream_name, nats_server, dequeue_timeout)
```

Comment:
Maybe just "NatsQueue"? In case we port it to Zig, you know. It should be transparent to the user.